### ISSUE TITLE
Overridden no-member causes pylint to ignore an assignment

### ISSUE THREAD

**Issue by dmtucker** [PyCQA/pylint#1118]
### Steps to reproduce

``` python
try:
  raise ValueError()
except ValueError as ex:
  print(ex)
try:
  pass
except pyVmomi.vim.fault.InvalidLogin as ex:  # pylint: disable=no-member
  print(ex.msg)
```
### Current behavior

```
[E1101(no-member), main] Instance of 'ValueError' has no 'msg' member
```
### Expected behavior

(no error)
- Note: The expected behavior should be the same as when `ex` is not reused:
  
  ``` python
  try:
    raise ValueError()
  except ValueError as exc:
    print(exc)
  try:
    pass
  except pyVmomi.vim.fault.InvalidLogin as ex:  # pylint: disable=no-member
    print(ex.msg)
  ```
### pylint --version output

```
pylint 1.6.4, 
astroid 1.4.8
Python 2.7.9 (default, Jun 29 2016, 13:08:31) 
[GCC 4.9.2]
```

```
pylint 1.6.4, 
astroid 1.4.8
Python 3.4.2 (default, Oct  8 2014, 10:45:20) 
[GCC 4.9.1]
```

**Comment by jacobtylerwalls** [comment]
Can't reproduce on 2.14 dev:
```python
% cat z.py
import enum
try:
  raise ValueError()
except ValueError as ex:
  print(ex)
try:
  pass
except enum.TYPO as ex:  # pylint: disable=no-member
  print(ex.msg)

% py -m pylint z

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 3.33/10, +6.67)
```
